### PR TITLE
Cherry-pick to 5.4 branch: [WASM] Fix KeyPath optimization bug around signature match

### PR DESCRIPTION
**Pull request by kateinoigakukun**
ref: https://github.com/swiftwasm/swift/pull/3334

### CODE REVIEW

**Comment by MaxDesiatov** [comment]
Do we have CI set up for 5.4 at all? Looks like no checks are going to run here

File: lib/SILOptimizer/Utils/KeyPathProjector.cpp [hunk 1]
```diff
@@ -237,7 +237,22 @@ class GettablePropertyProjector : public ComponentProjector {
       assert(getter->getArguments().size() == 2 + target.isOSBinFormatWasm());
       
       auto ref = builder.createFunctionRef(loc, getter);
-      builder.createApply(loc, ref, subs, {addr, parentValue});
+
+      std::vector<SILValue> args{addr, parentValue};
+      // FIXME(wasm): For wasm, KeyPath getter always take indices parameter
```
Comment (MaxDesiatov) [inline lib/SILOptimizer/Utils/KeyPathProjector.cpp:242]:
```suggestion
      // FIXME(wasm): For wasm, KeyPath getter always takes indices parameter
```

**Review by MaxDesiatov — APPROVED** [review]
LGTM

**Comment by kateinoigakukun** [comment]
OK, ready to land!

**Comment by MaxDesiatov** [comment]
Sure, feel free to merge 🙂

**Pull request merged**